core/src/main/java/org/apache/iceberg/BaseRewriteManifests.java

```diff
@@ -53,7 +53,7 @@
   private static final String REPLACED_MANIFESTS_COUNT = "manifests-replaced";
   private static final String PROCESSED_ENTRY_COUNT = "entries-processed";
 
-  private static final Set<ManifestEntry.Status> ALLOWED_ENTRY_STATUSES = ImmutableSet.of(
+  private static final ImmutableSet<ManifestEntry.Status> ALLOWED_ENTRY_STATUSES = ImmutableSet.of(
```

Comment:
Why does this need to change?

Reply:
```
/Users/fokkodriesprong/Desktop/incubator-iceberg/core/src/main/java/org/apache/iceberg/BaseRewriteManifests.java:56: warning: [MutableConstantField] Constant field declarations should use the immutable type (such as ImmutableList) instead of the general collection interface type (such as List)
  private static final Set<ManifestEntry.Status> ALLOWED_ENTRY_STATUSES = ImmutableSet.of(
```